File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/dialog/FirstTimeDialogWindow.java
package com.dansoftware.libraryapp.gui.firsttime.dialog;

import com.dansoftware.libraryapp.gui.util.LibraryAppStage;
import javafx.stage.Modality;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link javafx.stage.Stage} that is used for showing a {@link FirstTimeDialog}.
 *
 * @author Daniel Gyorffy
 */
public class FirstTimeDialogWindow extends LibraryAppStage {

    public FirstTimeDialogWindow(@NotNull FirstTimeDialog firstTimeDialog) {
        super("window.firsttime.title", firstTimeDialog);
        initModality(Modality.APPLICATION_MODAL);
        setAlwaysOnTop(true);
        setResizable(true);
    }
}

File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/dialog/FirstTimeDialogWindow.kt
package com.dansoftware.libraryapp.gui.firsttime.dialog

import com.dansoftware.libraryapp.gui.window.BaseWindow
import javafx.stage.Modality

/**
 * The [javafx.stage.Stage] that is used for showing a [FirstTimeDialog].
 *
 * @author Daniel Gyorffy
 */
class FirstTimeDialogWindow(firstTimeDialog: FirstTimeDialog) :
    BaseWindow<FirstTimeDialog>("window.firsttime.title", firstTimeDialog) {
    init {
        initModality(Modality.APPLICATION_MODAL)
        isAlwaysOnTop = true
        isResizable = true
    }
}